### ISSUE TITLE
Possible padding issue in `input['node_pos']` (and similar)

### ISSUE THREAD

**Issue by liziyu403** [pengwei07/AeroGTO#4]
Hi, thanks for the great repo!

I ran into a small issue during training — some tensor sizes didn’t match expectations.  
I checked the line in `utils.py` :

```python
torch.cat([tensor, torch.zeros(N_max - N + 1, S)], dim=-2)
```
and wondered if the ` +1`  might not be necessary.
eg., if N_max = 5 and N = 2, we’d expect 3 padding rows, but this adds 4.

There seem to be a few similar cases elsewhere as well.
Maybe I misunderstood something — just wanted to raise it in case it's helpful.

Thanks again :)

**Comment by small-dumpling** [comment]
Hi ,Thanks for your excellent and detailed question about the +1 in our padding logic: 

```python
torch.cat([tensor, torch.zeros(N_max - N + 1, S)], dim=-2). 
```
This is a very insightful point!
The +1 was indeed an intentional safeguard. Our primary concern was to prevent potential errors when ``N == N_max``. In this scenario, ``N_max - N`` would be 0, leading to ``torch.zeros(0, S)``. We were initially cautious that concatenating with such a zero-dimension tensor might cause issues or errors with torch.cat in some PyTorch versions. Therefore, using ``N_max - N + 1`` ensures that at least one row of zeros is always created for the concatenation, which we considered a safer approach.
We've actually re-tested this based on your query, and on PyTorch 2.3, removing the +1 (i.e., using ``N_max - N``) works correctly even when ``N == N_max``; torch.cat handles the concatenation with torch.zeros(0, S) without issues. However, we are not entirely certain if this robust behavior is consistent across all older PyTorch versions that users of this repository might be employing.
Given this, retaining the +1 is a conservative choice we made to ensure broader compatibility and prevent potential errors for users on different PyTorch versions. To handle the fact that this might create an ``N_max + 1`` dimension when ``N == N_max``, the tensor's original intended dimension (N_max) is restored in a subsequent processing step.
Your detailed question was very helpful in prompting us to revisit and clarify this logic. We really appreciate you taking the time to look so closely at the code!
If you have any other questions or observations, please feel free to ask. We're always happy to discuss and provide explanations.
Thanks again!
